desktop-widgets/command_divelist.cpp

```diff
@@ -802,6 +803,57 @@ SplitDiveComputer::SplitDiveComputer(dive *d, int dc_num) : SplitDivesBase(d, sp
 	setText(tr("split dive computer"));
 }
 
+MoveDiveComputerToFront::MoveDiveComputerToFront(dive *d, int dc_num)
+{
+	setText(tr("move dive computer to front"));
+
+	diveToRemove.dives.push_back(d);
+
+	dive *new_dive = make_first_dc(d, dc_num);
+	if (!new_dive)
+		return;
```

Comment:
wouldn't this deserve some error? Because you've now deleted the existing dive (which isn't really the same as moving dive computers within the dive...)

Reply:
Yes, this is unfortunate, but harmless:

Note that the old dive is added to a dive-to-remove list. But if we don't add an element to the dive-to-add list, the `workToBeDone()` function will return false and the command will never be executed.

This is solved in a later commit (the if is moved to the front), but I will also fix it here.

Reply:
thx